model/host/host.go

```diff
@@ -3511,14 +3512,31 @@ func (h *Host) UnsetPersistentDNSInfo(ctx context.Context) error {
 	return nil
 }
 
-// SetSleepSchedule updates the host's sleep schedule. Note that if some fields
+// SetSleepSchedule sets the sleep schedule for a given host
+func SetSleepSchedule(ctx context.Context, hostId string, schedule SleepScheduleInfo) error {
```

Comment:
we can make this private now if its only called within model/host!

Reply:
I'd also recommend moving this function to model/host/db.go (we try to keep DB updates there, and functions that use those DB updates here, so it makes sense to keep the Update function in this file. We aren't always consistent about this but generally its meant to help with organization)